lib/index.js

```diff
@@ -403,36 +460,66 @@ exports.getFunctionUrl = async (ctx, boundaryId, functionId) => {
 
 exports.getStorageClient = (ctx) => {
     const expiry = 60 * 16; // 15+1 min to align with Lambda lifecycle plus some buffer
-    const accessToken = Jwt.sign({}, Buffer.from(ctx.configuration.fusebit_storage_key, 'base64').toString('utf8'), {
-        algorithm: 'RS256',
-        expiresIn: expiry,
-        audience: ctx.configuration.fusebit_storage_audience,
-        issuer: ctx.configuration.fusebit_storage_issuer_id,
-        subject: ctx.configuration.fusebit_storage_subject,
-        keyid: ctx.configuration.fusebit_storage_key_id,
-        header: { jwtId: Date.now().toString() },
-    });
+    let accessToken = ctx.fusebit && ctx.fusebit.functionAccessToken;
+    if (ctx.configuration.fusebit_storage_key) {
+        // Legacy mode - use pre-creaeted issuer to mint the access token
+        accessToken = Jwt.sign({}, Buffer.from(ctx.configuration.fusebit_storage_key, 'base64').toString('utf8'), {
+            algorithm: 'RS256',
+            expiresIn: expiry,
+            audience: ctx.configuration.fusebit_storage_audience,
+            issuer: ctx.configuration.fusebit_storage_issuer_id,
+            subject: ctx.configuration.fusebit_storage_subject,
+            keyid: ctx.configuration.fusebit_storage_key_id,
+            header: { jwtId: Date.now().toString() },
+        });
+    }
 
     const url = generateStorageUrl(ctx.configuration);
 
-    return {
+    const isHierarchicalStorageSupported = accessToken === ctx.fusebit.functionAccessToken;
+
+    const ensureHierarchicalStorageSupported = () => {
+        if (!isHierarchicalStorageSupported) {
+            throw new Error('Hierarchical storage is not supported in this function.');
+        }
+    }
+
+    const getUrl = (storageSubId) => `${url}${storageSubId ? ('/' + storageSubId) : ''}`;
+
+    const storageClient = {
         etag: null,
         expiration: Date.now() + expiry * 1000,
-        get: async function () {
-            const response = await Superagent.get(url)
+        get: async function (storageSubId) {
```

Comment:
Why support this, instead of expecting the full id to be provided consistently?

Reply:
Using subId is the only scenario I am aware of for this client. Not having to think about constructing the whole thing is just simpler when I want to use subordinate documents. 

I expect we will have a proper, stand-alone SDK for all of fusebit APIs one day where it will make sense to support the full path, but I don't this is it.

Reply:
Check the new implementation